src/main/scala/s3dsl/S3Dsl.scala

```diff
@@ -130,16 +133,22 @@ object S3Dsl {
       // Object
       //
 
-      override def getObject(path: Path, chunkSize: Int): F[Option[Object[F]]] = for {
-        s3object <- F.blocking[Option[S3Object]](Some(s3.getObject(path.bucket.value, path.key.value))).handle404(None)
-        obj <- s3object.traverse { o =>
-          val isT: F[InputStream] = F.blocking(o.getObjectContent)
-          F.delay(Object[F](
-            stream = fs2.io.readInputStream[F](isT, chunkSize, blocker, closeAfterUse = true)(F, cs),
-            meta = toMeta(o.getObjectMetadata))
-          )
-        }
-      } yield obj
+      override def getObject(path: Path, chunkSize: Int): Resource[F, Stream[F, Byte]] = {
```

Comment:
This codomain can be simplified to `Stream[F, Byte]` by using `Stream.resource` combinator.